pkg/machine/wsl/machine.go

```diff
@@ -136,9 +136,56 @@ Wants=network-online.target podman.socket
 ExecStart=/usr/bin/sleep infinity
 `
 
-const lingerSetup = `mkdir -p /home/[USER]/.config/systemd/[USER]/default.target.wants
-ln -fs /home/[USER]/.config/systemd/[USER]/linger-example.service \
-       /home/[USER]/.config/systemd/[USER]/default.target.wants/linger-example.service
+const lingerSetup = `mkdir -p /home/[USER]/.config/systemd/user/default.target.wants
+ln -fs /home/[USER]/.config/systemd/user/linger-example.service \
+       /home/[USER]/.config/systemd/user/default.target.wants/linger-example.service
+`
+
+const bindMountSystemService = `
+[Unit]
+Description=Bind mount for system podman sockets
+After=podman.socket
+
+[Service]
+RemainAfterExit=true
+Type=simple
+# Ensure user services can register sockets as well
+ExecStartPre=mkdir -p -m 777 /mnt/wsl/podman-sockets/%[1]s
+ExecStartPre=touch /mnt/wsl/podman-sockets/%[1]s/podman-root.sock
+ExecStart=mount --bind %%t/podman/podman.sock /mnt/wsl/podman-sockets/%[1]s/podman-root.sock
+ExecStop=umount /mnt/wsl/podman-sockets/%[1]s/podman-root.sock
+`
+
+const bindMountUserService = `
+[Unit]
+Description=Bind mount for podman sockets
+After=podman.socket
+
+[Service]
+RemainAfterExit=true
+Type=simple
+# Consistency with system service (supports racing)
+ExecStartPre=mkdir -p -m 777 /mnt/wsl/podman-sockets/%[1]s
+ExecStartPre=touch /mnt/wsl/podman-sockets/%[1]s/podman-user.sock
+# Relies on /etc/fstab entry for user mounting
+ExecStart=mount /mnt/wsl/podman-sockets/%[1]s/podman-user.sock
+ExecStop=umount /mnt/wsl/podman-sockets/%[1]s/podman-user.sock
+`
+
+const bindMountFsTab = `/run/user/1000/podman/podman.sock /mnt/wsl/podman-sockets/%s/podman-user.sock none noauto,user,bind,defaults 0 0
+`
+
+const configBindServices = `mkdir -p /home/[USER]/.config/systemd/user/default.target.wants
+mkdir -p /etc/systemd/system/default.target.wants
+mkdir -p /etc/systemd/system/podman.socket.d/
+ln -fs /home/[USER]/.config/systemd/user/podman-mnt-bindings.service /home/[USER]/.config/systemd/user/default.target.wants/podman-mnt-bindings.service
+ln -fs /etc/systemd/system/podman-mnt-bindings.service /etc/systemd/system/default.target.wants/podman-mnt-bindings.service
```

Comment:
can we somehow make the paths into constants so that we can match them with configureBindMounts() below.
Otherwise someone might forget to change it.

Second, why to we use `[USER]` here compared to the normal sprintf() style? This seems rather inconstant and hard to follow.

Reply:
Sure I can improve that